vehicle/mkv/software/dashboard/dashboard.c

```diff
@@ -1,521 +1,144 @@
-void initTimer(void) {
-	TCCR0A = _BV(WGM01);    // Set up 8-bit timer in CTC mode
-	TCCR0B = 0x05;          // clkio/1024 prescaler
-	TIMSK0 |= _BV(OCIE0A);
-	OCR0A = 0x27;           //Makes timer run at ~100Hz
-}
-
-void initIO(void) {
-	/* Initialize all inputs and outputs and interrupts */
-
-	// Enable Interrupts
-	sei();
-
-  //Sets each of these pins as Outputs
-	DDRB |= _BV(DEBUG_LED1) | _BV(DEBUG_LED2) | _BV(IMD_LED) | _BV(START_LED) | _BV(BRAKE_LED) ;
-	DDRC |= _BV(RTD_LSD) | _BV(RJ_LED1) | _BV(RJ_LED2);//| _BV(START_PIN) --> Included below
-	DDRD |= _BV(BMS_LED) | _BV(HV_LED) | _BV(LV_LED);//| _BV(STEERING_POT) --> Included below
-
-  //Set the BMS LED as OFF
-	PORT_BMS_LED &= ~_BV(BMS_LED);
-
-	//Set start pin as input
-	DDRC &= ~_BV(START_PIN);
-
-	//Set Start Pin as input pull up, Turning it on
-	PORT_START |= _BV(START_PIN);
-
-	//Set pull up resistor for steering pot
-	PORT_STEERING_POT |= _BV(STEERING_POT);
-
-	//PORTC |= _BV(PC0);
-
-	/* Setup pin change interrupt registers for start pin*/
-	PCICR |= _BV(PCIE1) | _BV(PCIE2);
-	PCMSK1 |= _BV(PCINT14);
-	PCMSK2 |= _BV(PCINT18);
-
-
-
-	/*----- Setup PWM output -----*/
-	//Output compare pin is OC1B, so we need OCR1B as our counter
-	DDRC |= _BV(PC0); //Enable output pin
-	DDRC |= _BV(PC1); //Enable output pin
-
-	// DDRD |= _BV(PD3);
-	DDRD |= _BV(PD2); //PWM is happening on this pin
-
-	// pg 119 and 110 of datasheet
-	// currently running on mode 5 for 8-bit resolution timer
-	// turns on when counter hits OCRnx and off when hits top (255)
-	TCCR1A |= _BV(COM1A1) | _BV(COM1A0) | _BV(COM1B1) | _BV(COM1B0) | _BV(WGM10);
-	TCCR1A &= ~_BV(WGM11) & ~_BV(COM1A0) & ~_BV(COM1B0);
-	TCCR1B |= _BV(CS10); //Clock prescale set to max speed
-	TCCR1B |= _BV(WGM12);
-	TCCR1B &= ~_BV(WGM13);
-
-
-	// OCR1A = (uint8_t) 115;       // Duty Cycle
-	// OCR1B = (uint8_t) 127;      // Duty Cycle
-
+    Header:
+    Contains code for the dashboard board
+    Function: Responsible for AMS and IMD LED Indicators - DONE
+                -listening for CAN messages and controlling LED appropriately
+            : Start button + LED - DONE
+                - based on input off of the physical start button before
+   entering RTD : Interface with LED Bars Boards - TODO
+                - listening for CAN messages and outputting over I2C
+            : Implement other LEDs (brake, LV, HV) - DONE
+                - CAN
+            : Implement power cycle logic - DONE
+            : Implement RTD Buzzer - DONE
+    Questions:
+        Air control message location for IMD
+*/
+
+#include "dashboard_config.h"
+#include "libs/adc/api.h"
+#include "libs/gpio/api.h"
+#include "libs/timer/api.h"
+#include "vehicle/mkv/software/dashboard/can_api.h"
+
+volatile bool START_BUTTON_STATE;
+volatile bool HV_STATE;
+volatile bool BRAKE_PRESSED;
+volatile bool READYTODRIVE;
+
+volatile bool send_can;
+volatile int buzzer_counter = 0;
+
+// Start Button interrupt & final ReadyToDrive check
+void pcint1_callback(void) {
```

Comment:
You should do `START_BUTTON_STATE = !!gpio_get_pin(START_BTN);` and then in your loop you can do `if (START_BUTTON_STATE && HV_STATE && BRAKE_PRESSED) {....`.